src/test/java/com/icthh/xm/ms/entity/service/CustomMetricsIntTest.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.annotation.Repeat;
+import org.springframework.test.context.transaction.BeforeTransaction;
+
+/**
+ * Test class for the XmEntityResource REST controller.
+ *
+ * @see XmEntityResource
+ */
+@Slf4j
+@WithMockUser(authorities = {"SUPER-ADMIN"})
+public class CustomMetricsIntTest extends AbstractSpringBootTest {
+
+    @Autowired
+    private TenantContextHolder tenantContextHolder;
+
+    @Mock
+    private XmAuthenticationContextHolder authContextHolder;
+
+    @Mock
+    private XmAuthenticationContext context;
+
+    @Autowired
+    private LepManager lepManager;
+
+    @Autowired
+    private MetricRegistry metricRegistry;
+
+    @Autowired
+    private CustomMetricsConfiguration customMetricsConfiguration;
+
+    @Autowired
+    private XmLepScriptConfigServerResourceLoader lepResourceLoader;
+
+    @BeforeTransaction
+    public void beforeTransaction() {
+        TenantContextUtils.setTenant(tenantContextHolder, "RESINTTEST");
+    }
+
+    @SneakyThrows
+    @Before
+    public void setup() {
+
+        TenantContextUtils.setTenant(tenantContextHolder, "RESINTTEST");
+
+        MockitoAnnotations.initMocks(this);
+        when(authContextHolder.getContext()).thenReturn(context);
+        when(context.getUserKey()).thenReturn(Optional.of("userKey"));
+
+        lepManager.beginThreadContext(ctx -> {
+            ctx.setValue(THREAD_CONTEXT_KEY_TENANT_CONTEXT, tenantContextHolder.getContext());
+            ctx.setValue(THREAD_CONTEXT_KEY_AUTH_CONTEXT, authContextHolder.getContext());
+        });
+
+    }
+
+    @After
+    public void tearDown() {
+        lepManager.endThreadContext();
+        tenantContextHolder.getPrivilegedContext().destroyCurrentContext();
+    }
+
+    @SneakyThrows
+    public static String loadFile(String path) {
+        InputStream cfgInputStream = new ClassPathResource(path).getInputStream();
+        return IOUtils.toString(cfgInputStream, UTF_8);
+    }
+
+    @Test
+    @SneakyThrows
+    public void testCustomMetrics() {
+
+        lepResourceLoader.onRefresh("/config/tenants/RESINTTEST/entity/lep/metrics/Metric$$my$every$call$metric$$around.groovy",
+                                    loadFile("config/testlep/metrics/Metric$$my$every$call$metric$$around.groovy"));
+        lepResourceLoader.onRefresh("/config/tenants/RESINTTEST/entity/lep/metrics/Metric$$my$periodic$metric$$around.groovy",
+                                    loadFile("config/testlep/metrics/Metric$$my$periodic$metric$$around.groovy"));
+
+        customMetricsConfiguration.onRefresh("/config/tenants/RESINTTEST/entity/metrics.yml",
+                                             loadFile("config/metrics.yml"));
+
+        Gauge everyTimeMetric = metricRegistry.getGauges().get("resinttest.metrics.my.every.call.metric");
```

Comment:
please consider to change fomat of metric name from `resinttest.metrics.my.every.call.metric`

to 
custom.metrics.<tenant>.my.every.call

Reply:
resinttest it's name of tenant